### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by jaitor1** [3liz/lizmap-web-client#3152]
### What is the bug?

When using auth URL params (auth_login, url_password and auth_url_return) redirections not working properly.

I created a test map and dumb user/pass for this test. I modified the login form and added the allowAnyOrigin="true" to be able to log in via url parms. Don't worry about security, it is only used within a local network, in a controlled environment:

[http://eudala2.getxo.eus/lizmap/www/admin.php/auth/login/in?&auth_login=borrar&auth_password=borrar_test&auth_url_return=%2Flizmap%2Fwww%2Findex.php%2Fview%2Fmap%2F%3Frepository%3Dtest%26project%3Dtest](url)

1. First time you click on the link you get redirected to the login website. It does not log in although I have already entered the user and pass via URL params and it does not redirect to the specified map neither. I'm not sure if this particular misbehavior is a lizmap issue or a security misconfiguration in the administration I work for.

2. But second time you click on the link you get logged in correctly and redirected to the map properly.

3. Third time + you are already logged in correctly but you get always redirected to the admin website instead of to the map. 

In short, it only works the second time you hit the link.

### Steps to reproduce the issue

Already specified in bug description.

### Versions

- Lizmap version: 3.3.3
- QGIS Server FULL version: 3.10.10 LTR
- QGIS Desktop FULL version: 3.10.10 LTR
- OS (Windows, Linux, MacOS, Android…): Windows Server 2016 Standard.
- Lizmap admin information (users, logs…)? Everything default

### Check Lizmap plugin

- [X] I have done the step just before in the Lizmap QGIS desktop plugin before opening this ticket. Otherwise, my ticket is not considered valid and might get closed.

### QGIS server version, only if the section above doesn't mention the QGIS Server version

3.10

### Operating system

Windows Server 2016

### Browsers

Firefox, Chrome, Safari, Microsoft Edge

### Browsers version

90

**Comment by Gustry** [comment]
> Lizmap version: 3.3.3

This is not a supported version.
Can you try with latest 3.5.5 ?

3.3.3 is very old, released in december 2019 ...
https://github.com/3liz/lizmap-web-client/releases/tag/3.3.3

**Comment by jaitor1** [comment]
I installed version 3.5.5 on my server and I get the exact same behaviour.

You can check it yourself:
[http://eudala2.getxo.eus/lizmap35/admin.php/auth/login/in?&auth_login=borrar&auth_password=borrar_test&auth_url_return=%2Flizmap35%2Findex.php%2Fview%2Fmap%2F%3Frepository%3Dtest%26project%3Dtest](url)

**Comment by nworr** [comment]
After some test it seems that the url return param generate a redirection in all cases (user logged or not) when used in the auth/login/ url (not /in which is uses to process login form)
In my case the following Url worked in lizmap 3.5 , can you try it ?

http://eudala2.getxo.eus/lizmap35/admin.php/auth/login/?&auth_login=borrar&auth_password=borrar_test&auth_url_return=%2Flizmap35%2Findex.php%2Fview%2Fmap%2F%3Frepository%3Dtest%26project%3Dtest

in old version (3.3) behaviour is still the same with le login/ url

**Comment by jaitor1** [comment]
> After some test it seems that the url return param generate a redirection in all cases (user logged or not) when used in the auth/login/ url (not /in which is uses to process login form) In my case the following Url worked in lizmap 3.5 , can you try it ?
> 
> http://eudala2.getxo.eus/lizmap35/admin.php/auth/login/?&auth_login=borrar&auth_password=borrar_test&auth_url_return=%2Flizmap35%2Findex.php%2Fview%2Fmap%2F%3Frepository%3Dtest%26project%3Dtest
> 
> in old version (3.3) behaviour is still the same with le login/ url

@nworr thanks for your input.

Seems like in v3.5 the url you are giving (/auth/login?) redirects ok to map once you are logged in. But it is not different from going directly to the map (http://eudala2.getxo.eus/lizmap35/index.php/view/map/?repository=test&project=test) as you are already logged in. The url (/auth/login?) does not log in so does not really solve the problem.

A working URL should loginn and after that redirect to the map, all in one unique URL. If you are already logged in, just redirect to the map.

In v.3.3, as you mention, (/auth/login?) is not even redirecting to the map after you are logged in.

**Comment by nworr** [comment]
According to me, there can't be a unique URL with the desired behaviour, 
When not authenticated the aut_* params are used to perform a non-interactive login and  redirect to a provided url , but once the user is authenticated, the login form is out of sense , so all de log in process is ignored.

Maybe a trick can be made using a public project and a custom javascript wich detect if user is connected and redirect browser to the project URL or the login URL with auth_* params

**Comment by jaitor1** [comment]
> This issue is missing some feedbacks. 👻 Please have a look to the discussion, thanks. 🦎

Can someone remove the feedback tag? I have already provided feedback

**Comment by jaitor1** [comment]
With the help of ChatGPT, I modified the code in \lib\jelix-modules\jauth\controllers\login.classic.php to adapt it to my needs.

Please note that although we plan to update our entire infrastructure this year (including the Lizmap version), this modification is still for Lizmap 3.3.3.

You can check in the url provided that now it works.

I post it just in case someone needs it.

NEW \lib\jelix-modules\jauth\controllers\login.classic.php:

```
<?php
/**
* @package     jelix-modules
* @subpackage  jauth
* @author      Laurent Jouanneau
* @contributor Antoine Detante
* @copyright   2005-2008 Laurent Jouanneau, 2007 Antoine Detante
* @link        http://www.jelix.org
* @licence  http://www.gnu.org/licenses/lgpl.html GNU Lesser General Public Licence, see LICENCE file
*/

class loginCtrl extends jController {

    public $pluginParams = array(
      '*'=>array('auth.required'=>false)
    );

    function index()
    {
        if (jAuth::isConnected()) {
            // Si el usuario ya está conectado, se intenta redirigir
            $conf = jApp::coord()->getPlugin('auth')->config;

            $url = '';
            if ($conf['after_login'] != '') {
                $url = jUrl::get($conf['after_login']);
            }
            $auth_url_return = $this->param('auth_url_return');
            if ($auth_url_return != '') {
                // Se omite checkReturnUrl para usar siempre el valor pasado
                $url = $auth_url_return;
            }

            if ($url != '') {
                $rep = $this->getResponse('redirectUrl');
                $rep->url = $url;
                return $rep;
            }
        }
        
        $response = 'html';
        if (isset(jApp::config()->jcommunity)) {
            $conf = jApp::config()->jcommunity;
            $response = (isset($conf['loginResponse'])?$conf['loginResponse']:'html');
        }

        $rep = $this->getResponse($response);
        $rep->title = jLocale::get('login.login.title');
        $rep->body->assignZone('MAIN','jcommunity~login', array('as_main_content'=>true));
        return $rep;
    }

    /**
    *
    */
    function in() {
        $rep = $this->getResponse('redirectUrl');
        $conf = jApp::coord()->getPlugin('auth')->config;

        if ($conf['after_login'] == '') {
            throw new jException ('jcommunity~login.error.no.auth_login');
        }
        if ($conf['after_logout'] == '') {
            throw new jException ('jcommunity~login.error.no.auth_logout');
        }

        // Si ya está conectado, redirige directamente usando auth_url_return si existe
        if (jAuth::isConnected()) {
            $auth_url_return = $this->param('auth_url_return');
            if ($auth_url_return != '') {
                $rep->url = $auth_url_return;
            } else {
                $rep->url = jUrl::get($conf['after_login']);
            }
            return $rep;
        }

        // Intentamos capturar credenciales vía formulario
        $authLogin    = $this->param('auth_login');
        $authPassword = $this->param('auth_password');

        if ($authLogin && $authPassword) {
            // Intento de login sin formulario (credenciales en URL)
            if (!jAuth::login($authLogin, $authPassword, false)) {
                // Login fallido: redirige al login con parámetro de retorno si existe
                $auth_url_return = $this->param('auth_url_return');
                if ($auth_url_return) {
                    $rep->url = jUrl::get('login:index', array('auth_url_return'=>$auth_url_return));
                } else {
                    $rep->url = jUrl::get('login:index');
                }
                return $rep;
            }
            else {
                // Login correcto: se destruye el formulario (si lo hubiera) y se redirige
                jForms::destroy('jcommunity~login');
                if ($conf['enable_after_login_override']) {
                    $url_return = $this->param('auth_url_return');
                    if ($url_return) {
                        $rep->url = $url_return;
                    }
                    else {
                        $rep->url = jUrl::get($conf['after_login']);
                    }
                }
                else {
                    $rep->url = jUrl::get($conf['after_login']);
                }
                return $rep;
            }
        }

        // Si no se pasaron credenciales en la URL, usamos la lógica del formulario
        $form = jForms::fill('jcommunity~login');
        if (!$form) {
            $rep->url = jUrl::get($conf['after_logout']);
            return $rep;
        }

        if (!jAuth::login($form->getData('auth_login'), $form->getData('auth_password'), $form->getData('auth_remember_me'))) {
            $form->setErrorOn('auth_login', jLocale::get('jcommunity~login.error'));
            $auth_url_return = $this->param('auth_url_return');
            if ($auth_url_return) {
                $rep->url = jUrl::get('login:index', array('auth_url_return'=>$auth_url_return));
            } else {
                $rep->url = jUrl::get('login:index');
            }
        } else {
            jForms::destroy('jcommunity~login');
            if ($conf['enable_after_login_override']) {
                $url_return = $this->param('auth_url_return');
                if ($url_return) {
                    $rep->url = $url_return;
                }
                else {
                    $rep->url = jUrl::get($conf['after_login']);
                }
            }
            else {
                $rep->url = jUrl::get($conf['after_login']);
            }
        }

        return $rep;
    }

    /**
    *
    */
    function out() {
        $rep = $this->getResponse('redirectUrl');
        jAuth::logout();
        $conf = jApp::coord()->getPlugin ('auth')->config;

        if ($conf['after_logout'] == '') {
            throw new jException ('jcommunity~login.error.no.auth_logout');
        }

        if (jApp::coord()->execOriginalAction()) {
            if ($conf['enable_after_logout_override']) {
                $url_return = $this->param('auth_url_return');
                if ($url_return) {
                    $rep->url = $url_return;
                }
                else {
                    $rep->url = jUrl::get($conf['after_logout']);
                }
            }
        }
        else {
            if ($conf['enable_after_login_override'] && $_SERVER['REQUEST_METHOD'] == 'GET') {
                $rep->url = jUrl::get($conf['after_logout'],
                                      array('auth_url_return'=> jUrl::getCurrentUrl()));
            }
            else {
                $rep->url = jUrl::get($conf['after_logout']);
            }
        }
        return $rep;
    }
}


```
